pkg/policy/verifier.go

```diff
@@ -0,0 +1,210 @@
+// Copyright 2023 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package policy
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	ociremote "github.com/sigstore/cosign/v2/pkg/oci/remote"
+	"github.com/sigstore/policy-controller/pkg/apis/config"
+	"github.com/sigstore/policy-controller/pkg/webhook"
+	webhookcip "github.com/sigstore/policy-controller/pkg/webhook/clusterimagepolicy"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"knative.dev/pkg/apis"
+)
+
+// Verifier is the interface for checking that a given image digest satisfies
+// the policies backing this interface.
+type Verifier interface {
+	// Verify checks that the provided reference satisfies the backing policies.
+	Verify(context.Context, name.Reference, authn.Keychain) error
+}
+
+// WarningWriter is used to surface warning messages in a manner that
+// is customizable by callers that's suitable for their execution
+// environment.
+type WarningWriter func(string, ...interface{})
```

Comment:
Maybe my it would help me below (re: comment about surfacing warnings as typed errors below) if the arguments were documented here? What's the string, and more importantly I suppose what the interfaces here can be.

Reply:
Ah sure, yeah this is the universal `Printf` signature 😉 

Will update in the next push.